### PR TITLE
Generate testcase timing files for use with shards

### PR DESCRIPTION
**Pull request by aaroncameron-wk**
#### Reason for change
We currently hardcode testcase timing values for generating conformance suite shards. The timing data is not generated in a consistent way.

#### Description of change
We can automate this by generating a JSON file with the timings and loading the timing data during shard runs.

#### Steps to Test
CI (Only effects HMRC, KVK NT16 initially)

**review**:
@Arelle/arelle

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: tests/integration_tests/validation/validation_util.py [hunk 1]
```diff
@@ -282,3 +288,33 @@ def get_conformance_suite_test_results_without_shards(
...
+        return {
+            str(k): float(v)
+            for k, v in data.items()
+        }
+
+
+def save_timing_file(config: ConformanceSuiteConfig, results: list[ParameterSet]) -> None:
+    timing: dict[str, float] = defaultdict(float)
+    for result in results:
+        if not result.id:
```
[the remark this answers is not in the document]
Comment (aaroncameron-wk) [reply tests/integration_tests/validation/validation_util.py:308]: Probably doesn't. This is mostly to enforce that it isn't `NoneType` for typing, I'm not sure what's preferred: a `cast`, an `assert`, or this.

File: arelle/Validate.py [hunk 2]
```diff
@@ -168,6 +170,7 @@ def validateTestcase(self, testcase):
         elif hasattr(testcase, "testcaseVariations"):
             for modelTestcaseVariation in testcaseVariationsByTarget(testcase.testcaseVariations):
                 # update ui thread via modelManager (running in background here)
+                startTime = time.time()
```
[the remark this answers is not in the document]
Comment (aaroncameron-wk) [reply arelle/Validate.py:173]: TIL

File: tests/integration_tests/validation/run_conformance_suites.py [hunk 3]
```diff
@@ -154,6 +154,8 @@ def run_conformance_suites(
                 offline=offline_option,
                 series=series_option,
             )
+            if log_to_file and not shards:
```
[the remark this answers is not in the document]
Comment (aaroncameron-wk) [reply tests/integration_tests/validation/run_conformance_suites.py]: Left over from an earlier implementation, but looks like it works with shards now.

File: tests/integration_tests/validation/validation_util.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (aaroncameron-wk) [reply tests/integration_tests/validation/validation_util.py]: Went with `assert`

**Review by austinmatherne-wk — APPROVED** [review]

**Pull request merged**